tm2/pkg/bft/abci/types/types.go

```diff
@@ -116,6 +117,17 @@ func (r ResponseBase) IsErr() bool {
 	return r.Error != nil
 }
 
+func (r ResponseBase) EncodeEvents() []byte {
+	if len(r.Events) == 0 {
+		return []byte("[]")
+	}
+	res, err := json.Marshal(r.Events)
+	if err != nil {
+		return nil
```

Comment:
Panic?

Reply:
fixed to throw panic

Reply:
@notJoon 
I think this is still returning `nil`, and not panicking as @thehowl suggested

Reply:
@zivkovicmilos I forgot to link the commit, this was fixed earlier.
https://github.com/gnolang/gno/pull/1653/commits/ed5f542139ebadb4f0f8789386f3cf0441695cda#diff-1fab0863f74ecec7fede4fb382d4c71de279fe5ffca6a7b8c3a17e372e928a27